File: lc-dc-sep/lc-dc-sep6/solution.py
# Problem: 3217. Delete Nodes From Linked List Present in Array

class Solution:
    def modifiedList(self, nums: List[int], head: Optional[ListNode]) -> Optional[ListNode]:
        s = set()
        for num in nums:
            s.add(num)

        res = ListNode(0)
        temp = res

        while(head):
            val = head.val
            head = head.next
            if(val not in s):
                temp.next = ListNode(val)
                temp = temp.next

        return res.next

File: lc-dc-sep/lc-dc-sep6/solution.cpp
// Problem: 3217. Delete Nodes From Linked List Present in Array

class Solution {
public:
    ListNode* modifiedList(vector<int>& nums, ListNode* head) {
        unordered_set<int> s;
        for(auto num: nums) s.insert(num);

        ListNode* res = new ListNode(0);
        ListNode* temp = res;

        while(head){
            int val = head->val;
            head = head->next;
            if(!s.count(val)){
                temp->next = new ListNode(val);
                temp = temp-> next;
            }
        }

        return res->next;
    }
};